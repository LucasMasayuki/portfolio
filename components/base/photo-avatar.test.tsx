import { render } from '@testing-library/react';
import { describe, expect, it } from 'vitest';
import PhotoAvatar from './photo-avatar';

describe.concurrent('PhotoAvatar component', () => {
  describe('when picture is not loaded', () => {
    it('should render the first letter of the name', () => {
      // When rendered
      const { container } = render(<PhotoAvatar name={'test'} />);

      // Then it should show all content
      expect(container.innerHTML).toMatchSnapshot();
    });
  });

  describe('when image is loaded', () => {
    it('should render the picture', () => {
      // When rendered
      const { container } = render(
        <PhotoAvatar name={'test'} picture="test.com" />
      );

      // Then it should show all content
      expect(container.innerHTML).toMatchSnapshot();
    });
  });
});
