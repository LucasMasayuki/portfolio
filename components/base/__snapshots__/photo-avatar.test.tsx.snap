// Vitest Snapshot v1

exports[`PhotoAvatar component > when image is loaded > should render the picture 1`] = `"<div class=\\"MuiAvatar-root MuiAvatar-circular css-1fx3jmg-MuiAvatar-root\\"><img src=\\"test.com\\" class=\\"MuiAvatar-img css-1pqm26d-MuiAvatar-img\\"></div>"`;

exports[`PhotoAvatar component > when picture is not loaded > should render the first letter of the name 1`] = `"<div class=\\"MuiAvatar-root MuiAvatar-circular MuiAvatar-colorDefault css-hivpe0-MuiAvatar-root\\">T</div>"`;
